enterprise/cli/licenses.go

```diff
@@ -136,6 +137,71 @@ func validJWT(s string) error {
 }
 
 func (r *RootCmd) licensesList() *clibase.Cmd {
+	type tableLicense struct {
+		ID         int32     `table:"id,default_sort"`
+		UUID       uuid.UUID `table:"uuid" format:"uuid"`
+		UploadedAt time.Time `table:"uploaded_at" format:"date-time"`
+		// Features is the formatted string for the license claims.
+		// Used for the table view.
+		Features  string    `table:"features"`
+		ExpiresAt time.Time `table:"expires_at" format:"date-time"`
+		Trial     bool      `table:"trial"`
+	}
+
+	formatter := cliui.NewOutputFormatter(
+		cliui.ChangeFormatterData(
+			cliui.TableFormat([]tableLicense{}, []string{"UUID", "Expires At", "Uploaded At", "Features"}),
+			func(data any) (any, error) {
+				list, ok := data.([]codersdk.License)
+				if !ok {
+					return nil, xerrors.Errorf("invalid data type %T", data)
+				}
+				out := make([]tableLicense, 0, len(list))
+				for _, lic := range list {
+					var formattedFeatures string
+					features, err := lic.FeaturesClaims()
+					if err != nil {
+						formattedFeatures = xerrors.Errorf("invalid license: %w", err).Error()
+					} else {
+						var strs []string
+						if lic.AllFeaturesClaim() {
+							strs = append(strs, "all")
+						}
+						for k, v := range features {
+							strs = append(strs, fmt.Sprintf("%s=%v", k, v))
+						}
+						formattedFeatures = strings.Join(strs, ", ")
+					}
+					// If this returns an error, a zero time is returned.
+					exp, _ := lic.ExpiresAt()
+
+					out = append(out, tableLicense{
+						ID:         lic.ID,
+						UUID:       lic.UUID,
+						UploadedAt: lic.UploadedAt,
+						Features:   formattedFeatures,
+						ExpiresAt:  exp,
+						Trial:      lic.Trail(),
+					})
+				}
+				return out, nil
+			}),
+		cliui.ChangeFormatterData(cliui.JSONFormat(), func(data any) (any, error) {
+			list, ok := data.([]codersdk.License)
+			if !ok {
+				return nil, xerrors.Errorf("invalid data type %T", data)
+			}
+			for i := range list {
+				humanExp, err := list[i].ExpiresAt()
+				if err == nil {
+					list[i].Claims[codersdk.LicenseExpiryClaim+"_human"] = humanExp.Format(time.RFC3339)
```

Comment:
I'd say both this and the change in default output would warrant a breaking change.

Reply:
Fair :+1: